test/test_array_ops.cpp

```diff
@@ -516,6 +516,86 @@ TEST(ArrayOps, QuantizeAndDequantizeV2x8xtruexfalse) {
   opexecuter.RunTest();
 }  // end of test op QuantizeAndDequantizeV2x8xtruexfalse
 
+// CPU only supports QuantizedConcat with DT_QINT32 and DT_QUINT8
+TEST(ArrayOps, QuantizedConcat) {
+  Scope root = Scope::NewRootScope();
+  int dim1 = 2;
+  int dim2 = 3;
+
+  Tensor A(DT_QUINT8, TensorShape({dim1, dim2}));
+  AssignInputValues<quint8>(A, {5, 1, 0, 1, 5, 100});
+
+  Tensor B(DT_QUINT8, TensorShape({dim1, dim2}));
+  AssignInputValues<quint8>(B, {0, 2, 4, 6, 8, 10});
+
+  Tensor C(DT_QUINT8, TensorShape({dim1, dim2}));
+  AssignInputValues<quint8>(C, {1, 3, 5, 7, 9, 50});
+
+  vector<int> static_input_indexes = {0, 4, 5, 6, 7, 8, 9};
+
+  // TODO: NG and TF results disagress when input mins/maxes vary
+  ops::QuantizedConcat R = ops::QuantizedConcat(
+      root, 1, {A, B, C}, {-1.0f, -1.0f, -1.0f}, {3.0f, 3.0f, 3.0f});
+
+  vector<DataType> output_datatypes = {DT_QUINT8, DT_FLOAT, DT_FLOAT};
+
+  std::vector<Output> sess_run_fetchoutputs = {R.output, R.output_min,
+                                               R.output_max};
+  OpExecuter opexecuter(root, "QuantizedConcat", static_input_indexes,
+                        output_datatypes, sess_run_fetchoutputs);
+
+  opexecuter.RunTest();
+}  // end of test op QuantizedConcat
+
+// Disabled because for varing min/max input
+// NGraph and TF results diagree
+// For this test case: TF result is [47 46 46 0 1 2 70 72 73][46 47 55 3 3 4 75
+// 76 106]
+// NG result is [1 0 0 0 1 2 1 3 5][0 1 20 2 3 4 7 9 50]
+TEST(ArrayOps, DISABLED_QuantizedConcatVaryingMinMax) {
+  Scope root = Scope::NewRootScope();
+  int dim1 = 2;
+  int dim2 = 3;
+
+  Tensor A(DT_QUINT8, TensorShape({dim1, dim2}));
+  AssignInputValues<quint8>(A, {5, 1, 0, 1, 5, 100});
+
+  Tensor B(DT_QUINT8, TensorShape({dim1, dim2}));
+  AssignInputValues<quint8>(B, {0, 2, 4, 6, 8, 10});
+
+  Tensor C(DT_QUINT8, TensorShape({dim1, dim2}));
+  AssignInputValues<quint8>(C, {1, 3, 5, 7, 9, 50});
+
+  vector<int> static_input_indexes = {0, 4, 5, 6, 7, 8, 9};
+
+  ops::QuantizedConcat R = ops::QuantizedConcat(
+      root, 1, {A, B, C}, {1.0f, -1.0f, 2.0f}, {2.0f, 4.0f, 10.0f});
+
+  vector<DataType> output_datatypes = {DT_QUINT8, DT_FLOAT, DT_FLOAT};
+
+  std::vector<Output> sess_run_fetchoutputs = {R.output, R.output_min,
+                                               R.output_max};
+  OpExecuter opexecuter(root, "QuantizedConcat", static_input_indexes,
+                        output_datatypes, sess_run_fetchoutputs);
+
+  // vector<Tensor> tf_outputs;
+  // opexecuter.ExecuteOnTF(tf_outputs);
+
+  // vector<Tensor> ng_outputs;
+  // opexecuter.ExecuteOnNGraph(ng_outputs);
+
+  // cout << "TF outputs " << endl;
+  // for(auto i : tf_outputs){
+  //   PrintTensorAllValues(i, 100);
+  // }
+
+  // cout << "NG outputs " << endl;
+  // for(auto i : ng_outputs){
+  //   PrintTensorAllValues(i, 100);
+  // }
+  opexecuter.RunTest();
+}  // end of test op QuantizedConcatVaryingMinMax
+
```

Comment:
These 2 unit tests are for `QuantizedConcat`. Is it possible to add an unit test for `QuantizedConcatV2` as well?

Reply:
There is no TF QuantizedConcatV2 API, I got error sth like TF doesn't have this op registered.